drivers/mtd_spi_eeprom/mtd_spi_eeprom.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2220 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     drivers_mtd_spi_eeprom
+ * @{
+ *
+ * @file
+ * @brief       Driver for using spi eeprom (like AT25xxx, M95xxx, 25AAxxx, 25LCxxx,
+ *              CAT25xxx & BR25Sxxx) as mtd
+ *
+ * @author      Johannes Koster <johannes.koster@ml-pa.com>
+ *
+ * @}
+ */
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+#include "mtd.h"
+#include "mtd_spi_eeprom.h"
+#include "at25xxx.h"
+#include "at25xxx_params.h"
+
+#include <inttypes.h>
+#include <errno.h>
+
+static int mtd_spi_eeprom_init(mtd_dev_t *mtd);
+static int mtd_spi_eeprom_read(mtd_dev_t *mtd, void *dest, uint32_t addr,
+                           uint32_t size);
+static int mtd_spi_eeprom_write(mtd_dev_t *mtd, const void *src, uint32_t addr,
+                            uint32_t size);
+static int mtd_spi_eeprom_erase(mtd_dev_t *mtd, uint32_t addr, uint32_t size);
+static int mtd_spi_eeprom_power(mtd_dev_t *mtd, enum mtd_power_state power);
+
+const mtd_desc_t mtd_spi_eeprom_driver = {
+    .init = mtd_spi_eeprom_init,
+    .read = mtd_spi_eeprom_read,
+    .write = mtd_spi_eeprom_write,
+    .erase = mtd_spi_eeprom_erase,
+    .power = mtd_spi_eeprom_power,
+};
```

Comment:
If you move this to the end of the file, you can save yourself the forward-declaration.